platforms/quorum/configuration/roles/create/crypto/constellation/tasks/nested_main.yaml

```diff
@@ -0,0 +1,32 @@
+# This tasks checks if transaction manager key is already present in the vault
+- name: Check tm key is present the vault
+  shell: |
+    vault read -field=tm.pub secret/{{ component_ns }}/crypto/{{ node.name }}/transaction
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  register: vault_tm_result
+  ignore_errors: yes
+
+# Task creates the build directory if it is does not exist
+- name: Create build directory
+  file:
+    path: build/{{ component_name }}/{{ node.name }}
+    state: directory
+    mode: '0755'
+    recurse: yes
+  when: vault_tm_result.failed == True
+
+- name: Generate Crypto for constellation
+  shell: |
+    cd build/{{ component_name }}/{{ node.name }}
+    printf '\n' | constellation-node --generatekeys=tm
```

Comment:
add setup/constellation-node as a meta dependency, that way setup/constellation-node does not have to be called explicitly